railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -231,6 +231,24 @@ def initialize(*args)
           raise Error, "Invalid value for --database option. Supported for preconfiguration are: #{DATABASES.join(", ")}."
         end
 
+        # Fail hard if node or yarn isn't detected on the system
+        if options[:webpack]
+          check_node
+          check_yarn
+        end
+
+        # --webpack uses webpacker, which uses action view helpers
+        # --api skips action view
+        if options[:webpack] && options[:api]
```

Comment:
The gem does two things - one is using Webpack to compile modern javascript code and two involves helpers to fetch and link them within Rails views. Perhaps with api option supplied together with webpack we can re-structure bits of webpacker gem so that it serve files directly from `public` directory.